docs/auth.md

```diff
@@ -0,0 +1,462 @@
+  give his permissions.
+
+- The client can't make the assumptions about when its tokens will work. The
+  tokens have no meaning for him (like cookies in a browser), they are just
+  something it got from the authorization server and can send with its
+  request. The access token can expire, the user can revoke them, etc.
+
+- OAuth 2.0 defines no cryptographic methods. But a developer that want to use
+  it will have to put her hands in that.
+
+If you want to learn OAuth 2 in details, I recommend the [OAuth 2 in Action
+book](https://www.manning.com/books/oauth-2-in-action).
+
+
+The cozy stack as an authorization server
+-----------------------------------------
+
+### GET /auth/login
+
+Display a form with a password field to let the user authenticates herself to
+the cozy stack.
+
+This endpoint accepts a `redirect` parameter. If the user is already logged
+in, she will be redirected immediately. Else, the parameter will be transfered
+in the POST. This parameter can only contain a link to an application
+installed on the cozy (thus to a subdomain of the cozy instance). To protect
+against stealing authorization code with redirection, the fragment is always
+overriden:
+
+```http
+GET /auth/login?redirect=https://contacts.cozy.example.org/foo?bar#baz HTTP/1.1
+Host: cozy.example.org
+Cookies: ...
+```
+
+**Note**: the redirect parameter should be URL-encoded. We haven't done that
+to make it clear what the path (`foo`), the query-string (`bar`), and the
+fragment (`baz`) are.
+
+```http
+HTTP/1.1 302 Moved Temporarily
+Location: https://contacts.cozy.example.org/foo?bar#_=_
+```
+
+If the `redirect` parameter is invalid, the response will be `400 Bad
+Request`. Same for other parameters, the redirection will happen only on
+success (even if OAuth2 says the authorization server can redirect on errors,
+it's very complicated to do it safely, and it is better to avoid this trap).
+
+### POST /auth/login
+
+After the user has typed her password and clicked on `Login`, a request is
+made to this endpoint.
+
+The `redirect` parameter is passed inside the body. If it is missing, the
+redirection will be made against the default target: the home application of
+this cozy instance.
+
+```http
+POST /auth/login HTTP/1.1
+Host: cozy.example.org
+Content-type: application/x-www-form-urlencoded
+
+password=p4ssw0rd&redirect=https%3A%2F%2Fcontacts.cozy.example.org
+```
+
+```http
+HTTP/1.1 302 Moved Temporarily
+Set-Cookie: ...
+Location: https://contacts.cozy.example.org/foo
+```
+
+### DELETE /auth/login
+
+This can be used to log-out the user. A private context must be passed in the
+query-string, to protect against CSRF attack on this (this can part of bigger
+attacks like session fixation).
+
+```http
+DELETE /auth/login?CtxToken=token-for-a-private-context HTTP/1.1
+Host: cozy.example.org
+```
+
+### POST /auth/register
+
+This route is used by OAuth2 clients to dynamically register them-selves.
```

Comment:
I am not sure I understand the added security from this step : 

- Either we start an actual cozy partnership program and dynamic registration requires something cryptosigned by the cozy cloud company or for the cozy to be in "dev" mode. In which case, it might be important to specify it here.
- If there is no such verification, this is (IMHO) **less** secure than simply accepting any redirect URL and warning the user : fishing is way easier with added description, ie difference between
`[LOGO Partner] The partner company SA  (https://almosttheprtnerurl.com)` vs `You are about to share data with https://almosttheprtnerurl.com, BE CAREFUL, make sure its the correct url, cozy can make no guarantee other than the url`